src/main/java/com/github/manolo8/darkbot/Bot.java

```diff
@@ -32,21 +43,83 @@ public static void main(String[] args) throws IOException {
         LibSetup.setupLibraries();
         StartupParams params = new StartupParams(args);
 
-        checkJavaVersion(params);
         SwingUtilities.invokeLater(() -> new Main(params));
     }
 
-    private static void checkJavaVersion(StartupParams params) {
+    static void checkUniqueInstance(StartupParams params) {
+        if (params.has(StartupParams.LaunchArg.NO_WARN)) return;
+
+        Path filePath = Paths.get("curr.pid");
+        long currentPid = ProcessHandle.current().pid();
+        ProcessHandle processHandle = ProcessHandle.current();
+        ProcessHandle.Info processInfo = processHandle.info();
+        long currentStartTime = processInfo.startInstant().map(Instant::toEpochMilli).orElse(0L);
+
+        List<String> fileContent;
+        try {
+            fileContent = Files.exists(filePath) ? Files.readAllLines(filePath, StandardCharsets.UTF_8) : Collections.emptyList();
```

Comment:
```suggestion
        try {
                    List<String> fileContent = Files.exists(filePath) ? 
                            Files.readAllLines(filePath, StandardCharsets.UTF_8) : Collections.emptyList();
```
doesn't seem like fileContent is used anywhere outside, so can likely just be put in the inner scope